src/ledctl.c

```diff
@@ -840,31 +725,26 @@ static ledctl_status_code_t list_slots(struct slot_request *slot_req)
  */
 ledctl_status_code_t slot_execute(struct slot_request *slot_req)
 {
-	struct slot_response slot_res;
-	ledctl_status_code_t status = LEDCTL_STATUS_SUCCESS;
+	struct slot_property *slot;
+
+	if (slot_req->chosen_opt == OPT_LIST_SLOTS)
+		return list_slots(slot_req->cntrl);
 
-	slot_response_init(&slot_res);
+	slot = find_slot(slot_req);
+	if (slot == NULL)
+		return LEDCTL_STATUS_DATA_ERROR;
 
 	switch (slot_req->chosen_opt) {
-	case OPT_LIST_SLOTS:
-		return list_slots(slot_req);
 	case OPT_SET_SLOT:
-		status = slot_req->get_slot_fn(slot_req->device, slot_req->slot, &slot_res);
-		if (status != LEDCTL_STATUS_SUCCESS)
-			return status;
-		if (slot_res.state == slot_req->state) {
+		if (get_slot_pattern(slot) == slot_req->state) {
 			log_warning("Led state: %s is already set for the slot.",
 				    ibpi2str(slot_req->state));
 			return LEDCTL_STATUS_SUCCESS;
 		}
-		status = slot_req->set_slot_fn(slot_res.slot, slot_req->state);
-		if (status != LEDCTL_STATUS_SUCCESS)
-			return status;
+		return set_slot_pattern(slot, slot_req->state);
```

Comment:
I've noticed that set-slot will be executed without any output when command ends successfully. Previously get-slot was executed also to print slot with new changed state.
It is ok for us, to not have output message when get-slot was executed without errors. Just let you know of this change.